barretenberg/cpp/src/barretenberg/vm2/simulation/ecc.hpp

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <cstdint>
+#include <memory>
+
+#include "barretenberg/vm2/common/memory_types.hpp"
+#include "barretenberg/vm2/simulation/context.hpp"
+#include "barretenberg/vm2/simulation/events/ecc_event.hpp"
+
+namespace bb::avm2::simulation {
+
+class EccInterface {
+  public:
+    virtual ~EccInterface() = default;
+    virtual void add(ContextInterface& context,
+                     MemoryAddress p_x_offset,
+                     MemoryAddress p_y_offset,
+                     MemoryAddress p_is_inf_offset,
+                     MemoryAddress q_x_offset,
+                     MemoryAddress q_y_offset,
+                     MemoryAddress q_is_inf_offset,
+                     MemoryAddress dst_x_offset,
+                     MemoryAddress dst_y_offset,
+                     MemoryAddress dst_is_inf_offset) = 0;
```

Comment:
I think if the circuit is not memory aware, then the simulation counterpart should not take memory.